src/sqlfluff/rules/L032.py

```diff
@@ -44,18 +56,130 @@ class Rule_L032(BaseRule):
 
     """
 
-    def _eval(self, context: RuleContext) -> Optional[List[LintResult]]:
+    def _eval(self, context: RuleContext) -> Optional[LintResult]:
         """Look for USING in a join clause."""
-        if context.segment.is_type("join_clause"):
-            for seg in context.segment.segments:
-                if seg.is_type("keyword") and seg.name == "using":
-                    return [
-                        LintResult(
-                            # Reference the element, not the string.
-                            anchor=seg,
-                            description=(
-                                "Found USING statement. Expected only ON statements."
-                            ),
-                        )
-                    ]
-        return None
+        segment = context.functional.segment
+        parent_stack = context.functional.parent_stack
+        # We are not concerned with non join clauses
+        if not segment.all(sp.is_type("join_clause")):
+            return None
+
+        using_anchor = segment.children(sp.is_keyword("using")).first()
+        # If there is no evidence of a USING then we exit
+        if len(using_anchor) == 0:
+            return None
+
+        anchor = using_anchor.get()
+        description = "Found USING statement. Expected only ON statements."
+        # All returns from here out will be some form of linting error.
+        # we prepare the variable here
+        unfixable_result = LintResult(
+            anchor=anchor,
+            description=description,
+        )
+
+        tables_in_join = parent_stack.last().children(
+            sp.is_type("join_clause", "from_expression_element")
+        )
+
+        # If we have more than 2 tables we wont try to fix join.
+        # TODO: if this is table 2 of 3 it is still fixable
+        if len(tables_in_join) > 2:
+            return unfixable_result
+
+        predicate = sp.is_type("select_statement")
+        parent_select = parent_stack.last(predicate).get()
```

Comment:
Think we could eliminate the variable?
```suggestion
        parent_select = parent_stack.last(sp.is_type("select_statement")).get()
```

Reply:
Black got upset with line length here :(

Reply:
👍